server/src/main/java/com/github/animelist/animelist/config/JacksonConfig.java

```diff
@@ -0,0 +1,17 @@
+package com.github.animelist.animelist.config;
+
+import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
+import org.bson.types.ObjectId;
+import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class JacksonConfig {
+
+    @Bean
+    public Jackson2ObjectMapperBuilderCustomizer jsonCustomizer() {
+        return builder -> builder.serializerByType(ObjectId.class, new ToStringSerializer());
```

Comment:
the Jackson serializer is used to serialize the object into a JSON document right? so if we're converting the objectID to a string again then how does it save space? not sure I fully understand. code lgtm though

Reply:
yes jackson is used to serialize the object into json. we're adding some more configuration to the jackson ObjectMapper SpringBoot already has for us through this configuration and bean file. it's used when we want to map the object to json whenever we want to return a api response.

mongodb does not use springboot's jackson ObjectMapper to convert the object to json. I think it uses another implementation, MongoTypeMapper, to do that which we can change and customize in the "mapping" layer [detailed here](https://docs.spring.io/spring-data/mongodb/docs/current/reference/html/#mongo-template.type-mapping)

in short, when mapping the object to store in MongoDB the objectId is correctly stored as an ObjectId. When mapping the object to json for an api response then the ObjectID is turned into a string.